packages/cli/src/executions/executions.controller.ts

```diff
@@ -1,31 +1,54 @@
 import { ExecutionRequest } from './execution.request';
-import { ExecutionsService } from './executions.service';
+import { ExecutionService } from './execution.service';
 import { Authorized, Get, Post, RestController } from '@/decorators';
-import { EnterpriseExecutionsService } from './executions.service.ee';
+import { EnterpriseExecutionsService } from './execution.service.ee';
 import { isSharingEnabled } from '@/UserManagement/UserManagementHelper';
+import { WorkflowSharingService } from '@/workflows/workflowSharing.service';
 
 @Authorized()
 @RestController('/executions')
 export class ExecutionsController {
+	constructor(
+		private readonly executionService: ExecutionService,
+		private readonly enterpriseExecutionService: EnterpriseExecutionsService,
+		private readonly workflowSharingService: WorkflowSharingService,
+	) {}
+
 	@Get('/')
 	async getExecutionsList(req: ExecutionRequest.GetAll) {
-		return ExecutionsService.getExecutionsList(req);
+		const sharedWorkflowIds = isSharingEnabled()
+			? await this.workflowSharingService.getSharedWorkflowIds(req.user)
+			: await this.workflowSharingService.getSharedWorkflowIds(req.user, ['owner']);
```

Comment:
```suggestion
		const sharedWorkflowIds = await this.workflowSharingService.getSharedWorkflowIds(
			req.user,
			isSharingEnabled() ? undefined : ['owner'],
		);
```

You could also extract this into a private method on this class, to get rid of the 3 other duplicates.